static/js/alert.js

```diff
@@ -818,12 +819,49 @@ function alertDetailsFunctions() {
     });
 }
 
-function createAlertFromLogs(queryLanguage, query, startEpoch, endEpoch) {
+function createAlertFromLogs(queryLanguage, searchText, startEpoch, endEpoch, filterTab) {
+    // Focus on the alert rule name input field
     $('#alert-rule-name').focus();
-    $('#query').val(query);
-    $(`.ranges .inner-range #${startEpoch}`).addClass('active');
+
+    // Set the query language and mark the appropriate option as active
+    $('#query-language-btn span').html(queryLanguage);
+    $('.query-language-option').removeClass('active');
+
+    if (queryLanguage === 'SQL') {
+        $('#option-1').addClass('active');
+    } else if (queryLanguage === 'Log QL') {
+        $('#option-2').addClass('active');
+    } else if (queryLanguage === 'Splunk QL') {
+        $('#option-3').addClass('active');
+    }
+    let filterValue = searchText;
+    if (filterTab == '0' || filterTab == null) {
+        if (filterValue != '*') {
+            codeToBuilderParsing(filterValue);
+            $('#filter-input').val(filterValue).change();
+            toggleClearButtonVisibility();
+            isQueryBuilderSearch = true;
+        }
+    } else {
+        $('#custom-code-tab').tabs('option', 'active', 1);
+        if (filterValue === '*') {
+            $('#filter-input').val('').change();
+        } else {
+            $('#filter-input').val(filterValue).change();
+            toggleClearButtonVisibility();
+        }
+        isQueryBuilderSearch = false;
+    }
```

Comment:
You can simplify this part.

```suggestion
if (filterTab === '0') {
    codeToBuilderParsing(searchText);
} else if (filterTab === '1') {
    $('#custom-code-tab').tabs('option', 'active', 1);
    $('#filter-input').val(searchText);
}
```